### PR TITLE
[twitter] Add expected field to parent item select in twitter_post_hook()

### PR DESCRIPTION
**Pull request by MrPetovan**
- Address https://github.com/friendica/friendica/issues/9250#issuecomment-751445732

### CODE REVIEW

File: twitter/twitter.php [hunk 1]
```diff
@@ -557,28 +559,24 @@ function twitter_post_hook(App $a, array &$b)
 		}
 
 		$condition = ['uri' => $b["thr-parent"], 'uid' => $b["uid"]];
-		$orig_post = Item::selectFirst(['author-nick', 'author-network'], $condition);
-		if (!DBA::isResult($orig_post)) {
+		$thr_parent = Item::selectFirst(['uri', 'author-nick', 'author-network'], $condition);
```
Comment (annando) [inline twitter/twitter.php]: Can you include `extid` to this as well? Seems like it is needed in line 610.

Comment (MrPetovan) [reply twitter/twitter.php]: Good catch!

File: twitter/twitter.php [hunk 2]
```diff
@@ -557,28 +559,24 @@ function twitter_post_hook(App $a, array &$b)
 		}
 
 		$condition = ['uri' => $b["thr-parent"], 'uid' => $b["uid"]];
-		$orig_post = Item::selectFirst(['author-nick', 'author-network'], $condition);
-		if (!DBA::isResult($orig_post)) {
+		$thr_parent = Item::selectFirst(['uri', 'extid', 'author-nick', 'author-network'], $condition);
```
Comment (annando) [inline twitter/twitter.php]: I realized that "author-link" is missing as well ... I hadn't checked that very well.

Comment (MrPetovan) [reply twitter/twitter.php]: It looks so much like `author-nick`!

**Pull request merged**